### PR TITLE
Feature/truck list

### LINKED ISSUE #25: As a User I can toggle to a list of trucks
- [x] When I view the map, I can see a button that says toggle list
- [x] when I click the toggle list button I will be shown a list of trucks and the map will be hidden
- [x] When I see a list of trucks I can see a button that says toggle map

### LINKED ISSUE #50: As A user I should be able to see a list of trucks in my radius
As a user, I should see a list of trucks with:

- [x] distance
- [x] name
- [x] description
- [x] logo

### PR DESCRIPTION
**Pull request by Shakikka**
## What does this PR do?

- Creates a list view of trucks
- displays trucks by closest distance
- Adds ability to toggle between map and truck list
- adds ability to click on truck card
- Adds dropdown menu
- Adds route for trucks by name

## How should it be tested?

- Go to the website and log in. Click on the truck list button, click on a truck card and note that the pathway changes.

## What do the future iterations hold?

- Add filtration ability for payment types in dropdown menu
- Clicking a truck card will eventually take you to a truck info page

## Linked Issues
- #25 
- #50

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -24,21 +25,26 @@ class App extends Component {
...
+        this.sortByDistance(trucks.data)
       })
     })
-    .catch(error => console.log(error))
+    .catch(error => this.setState({ error: error.message }))
+  }
+
+  sortByDistance = (trucks) => {
+    const sortedTrucks = trucks.sort((a, b) => a.attributes.distance - b.attributes.distance)
+    this.setState({ trucks: sortedTrucks })
```
Comment (elizhahn) [inline src/components/App.js:47]: could move this into a utility file in the future when all the filter methods are flushed out, we can rename this "filter trucks" and then use the sortByDistance method when distance is clicked

**Review by elizhahn — CHANGES_REQUESTED** [review]
We decided that we wanted testing to be on these PRs as well, moving in that direction I will wait to merge it so you can add testing first unless you want to start doing that on separate branches

File: src/components/App.js (hunk 1 above)
Comment (Shakikka) [reply src/components/App.js:47]: How would that work with setting state?

Comment (elizhahn) [reply src/components/App.js:47]: we would still set state within the filter function, and use the return of the helper functions

**Review by elizhahn — APPROVED** [review]

**Pull request merged**